api/processors/actions/group/finish.py

```diff
@@ -0,0 +1,76 @@
+from django.db import transaction
+
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.log_event_type import (
+    CoordinatorLogEventType,
+    GroupLogEventType,
+    StudentLogEventType,
+    TeacherLogEventType,
+)
+from api.models.choices.status import (
+    GroupProjectStatus,
+    StudentProjectStatus,
+    TeacherProjectStatus,
+)
+from api.processors.actions.group import GroupActionProcessor
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupFinishProcessor(GroupActionProcessor):
+    @transaction.atomic
+    def process(self) -> None:
+        self._set_statuses()
+        self._create_log_events()
+        self._move_related_people_to_former()
+
+    def _create_log_events(self) -> None:
+        GroupLogEventCreator.create(
+            group=self.group,
+            student_log_event_type=StudentLogEventType.GROUP_FINISHED,
+            teacher_log_event_type=TeacherLogEventType.GROUP_FINISHED,
+            coordinator_log_event_type=CoordinatorLogEventType.GROUP_FINISHED,
+            group_log_event_type=GroupLogEventType.FINISHED,
+            from_group=self.group,
+        )
+
+    def _move_related_people_to_former(self) -> None:
+        teachers_current, students_current, coordinators_current = (
+            self.group.teachers,
+            self.group.students,
+            self.group.coordinators,
+        )
+        self.group.teachers_former.add(*teachers_current.all())
+        self.group.students_former.add(*students_current.all())
+        self.group.coordinators_former.add(*coordinators_current.all())
+        self.group.teachers.clear()
+        self.group.students.clear()
+        self.group.coordinators.clear()
+        self.group.save()
```

Comment:
this is exactly the same in `abort.py`. Maybe this functionality could be factored out to the base class?

Reply:
This one indeed can be moved